src/NHibernate/Cfg/XmlHbmBinding/ClassBinder.cs

```diff
@@ -301,16 +301,11 @@ protected string GetClassTableName(PersistentClass model, string mappedTableName
 				model.ComponentClassName = FullQualifiedClassName(componentClassName, mappings);
 				model.IsEmbedded = false;
 			}
-			else if (componentMapping is HbmDynamicComponent)
-			{
-				model.IsEmbedded = false;
-				model.IsDynamic = true;
-			}
-
 			else if (reflectedClass != null)
 			{
 				model.ComponentClass = reflectedClass;
 				model.IsEmbedded = false;
+				model.IsDynamic = componentMapping is HbmDynamicComponent;
 			}
```

Comment:
Change to this file can be reverted if we do not care about "invalid" property declarations (see my [comment](https://github.com/nhibernate/nhibernate-core/pull/305/files#r193271703)).